src/model/openthaigpt_pretraining_model/tokenizers/spm_trainer.py

```diff
@@ -136,9 +145,17 @@ def train_tokenizer(
         sentence_iterator=iter(
             DataSetColumnIterator(text_processed_dataset, PREPARE_DATASETS_KEY)
         ),
-        model_prefix=output_path,
+        model_prefix=output_path + "/spm_tokenizer",
         vocab_size=vocab_size,
         user_defined_symbols=USER_DEFINED_SYMBOLS,
         num_threads=num_proc,
         train_extremely_large_corpus=large_corpus,
+        model_type=mode,
     )
+
+    tokenizer = LlamaTokenizer(vocab_file=output_path + "/spm_tokenizer.model")
```

Comment:
Can this script use with any tokenizer or just this tokenizer?

Reply:
Just this tokenizer